contractcourt/breacharbiter.go

```diff
@@ -47,6 +47,12 @@ var (
 	// procedure, we can recover and continue from the persisted state.
 	retributionBucket = []byte("retribution")
 
+	// taprootRetributionBucket stores the tarpoot specific retribution
+	// information. This includes things like the control blocks for both
+	// commitment outputs, and the taptweak needed to sweep each HTLC (one
+	// for the first and one for the second level).
+	taprootRetributionBucket = []byte("tap-retribution")
```

Comment:
> For encoding, if control block is nil, we don't write anything to disk. For decoding, we'd reach an EOF so no control block will be decoded.

So I don't think this quite works since all the contracts are encoded one after the other in the same buffer: https://github.com/lightningnetwork/lnd/blob/b565f4f5366dfaecfdcc5d7d682fd8bd6307226f/contractcourt/breacharbiter.go#L1634-L1638 

So we can't assume that the byte stream isn't shared amongst serialized breached outputs (possibility of the `EOF` check). 

For the nursery, each output is stored as a distinct value, rather than one value for all outputs. This allowed me to use the `EOF` trick there as we can assume that the byte stream "ends" after the full output has been serialized.